storage/innobase/xtrabackup/src/backup_mysql.cc

```diff
@@ -1873,6 +1873,11 @@ write_xtrabackup_info(MYSQL *connection)
 		"encrypted ENUM('Y', 'N') DEFAULT NULL"
 		") CHARACTER SET utf8 ENGINE=innodb", false);
 
+	/* Upgrade from previous versions */
+	xb_mysql_query(connection,
+		"ALTER TABLE PERCONA_SCHEMA.xtrabackup_history MODIFY COLUMN "
+		"binlog_pos TEXT DEFAULT NULL", false);
```

Comment:
can you please verify if ALTER is no-op if column is already in TEXT?

Reply:
@satya-bodapati , yes. See below:

```
mysql> CREATE TABLE `joinit` (
    ->   `i` int(11) NOT NULL AUTO_INCREMENT,
    ->   `s` varchar(64) DEFAULT NULL,
    ->   `t` time NOT NULL,
    ->   `g` int(11) NOT NULL,
    ->   PRIMARY KEY (`i`)
    -> ) ENGINE=InnoDB  DEFAULT CHARSET=latin1;
Query OK, 0 rows affected (0.02 sec)

-- INSERT 2M rows --

mysql> ALTER TABLE joinit MODIFY `s` TEXT DEFAULT NULL;
Query OK, 2097152 rows affected (1 min 35.44 sec)
Records: 2097152  Duplicates: 0  Warnings: 0

mysql> ALTER TABLE joinit MODIFY `s` TEXT DEFAULT NULL;
Query OK, 0 rows affected (0.01 sec)
Records: 0  Duplicates: 0  Warnings: 0
```

Reply:
@satya-bodapati yes, it doesn't change the column if the new type is equal to previous one:

```
mysql> CREATE TABLE `joinit` (
    ->   `i` int(11) NOT NULL AUTO_INCREMENT,
    ->   `s` varchar(64) DEFAULT NULL,
    ->   `t` time NOT NULL,
    ->   `g` int(11) NOT NULL,
    ->   PRIMARY KEY (`i`)
    -> ) ENGINE=InnoDB  DEFAULT CHARSET=latin1;
Query OK, 0 rows affected (0.02 sec)

-- Insert 2M rows --

mysql> ALTER TABLE joinit MODIFY `s` TEXT DEFAULT NULL;
Query OK, 2097152 rows affected (1 min 35.44 sec)
Records: 2097152  Duplicates: 0  Warnings: 0

mysql> ALTER TABLE joinit MODIFY `s` TEXT DEFAULT NULL;
Query OK, 0 rows affected (0.01 sec)
Records: 0  Duplicates: 0  Warnings: 0
```

Reply:
Thanks for checking